src/event.rs

```diff
@@ -44,12 +44,12 @@ impl Event {
     /// let event = Event::new_message(Level::Debug, Some("logger".into()), "test");
     /// event.capture();
     /// ```
-    pub fn new_message<S: Into<SentryString>>(
-        level: Level,
-        logger: Option<SentryString>,
-        text: S,
-    ) -> Self {
-        let logger = logger.map_or(ptr::null(), |logger| logger.as_cstr().as_ptr());
+    pub fn new_message<L, S>(level: Level, logger: Option<L>, text: S) -> Self
+    where
+        L: Into<SentryString>,
+        S: Into<SentryString>,
+    {
```

Comment:
Sadly that's a regression imo, while it improves the use of `Some`, if you try to use `None` it will tell you to define `L`, which would look like this: `None::<SentryString>`.

Maybe we get https://github.com/rust-lang/rust/issues/27336 one day.